### ISSUE TITLE
In e2e tests, wait on the timeout of waitForFunction/waitForSelector/etc instead of the Jest timeout

### ISSUE THREAD

**Issue by ijt** [sourcegraph/sourcegraph#5340]
This is based on @felixfbecker's [comment](https://github.com/sourcegraph/sourcegraph/pull/5302#discussion_r316847452) on a recent PR adding some e2e tests. All the tests should probably be changed based on his suggestion, not just the ones added in that PR.

**Comment by lguychard** [comment]
This is no longer relevant.
